sparse/impl/KokkosSparse_sptrsv_cuSPARSE_impl.hpp

```diff
@@ -60,6 +60,142 @@ void sptrsvcuSPARSE_symbolic(KernelHandle* sptrsv_handle,
                              ain_nonzero_index_view_type entries,
                              ain_values_scalar_view_type values, bool trans) {
 #ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+#if (CUDA_VERSION >= 11030)
+  typedef typename KernelHandle::nnz_lno_t idx_type;
+  typedef typename KernelHandle::size_type size_type;
+  typedef typename KernelHandle::scalar_t scalar_type;
+  typedef typename KernelHandle::memory_space memory_space;
+  typedef typename KernelHandle::nnz_scalar_view_t nnz_scalar_view_t;
+
+  const bool is_cuda_space =
+      std::is_same<memory_space, Kokkos::CudaSpace>::value ||
+      std::is_same<memory_space, Kokkos::CudaUVMSpace>::value ||
+      std::is_same<memory_space, Kokkos::CudaHostPinnedSpace>::value;
+
+  const bool is_idx_type_supported = std::is_same<idx_type, int>::value ||
+                                     std::is_same<idx_type, int64_t>::value;
+
+  if (!is_cuda_space) {
+    throw std::runtime_error(
+        "KokkosKernels sptrsvcuSPARSE_symbolic: MEMORY IS NOT ALLOCATED IN GPU "
+        "DEVICE for CUSPARSE\n");
+  } else if (is_idx_type_supported) {
+    bool is_lower = sptrsv_handle->is_lower_tri();
+    sptrsv_handle->create_cuSPARSE_Handle(trans, is_lower);
+
+    typename KernelHandle::SPTRSVcuSparseHandleType* h =
+        sptrsv_handle->get_cuSparseHandle();
+
+    cusparseStatus_t status;
+
+    int64_t nnz = static_cast<int64_t>(entries.extent(0));
+    size_t pBufferSize;
+    void* rm;
+    // NOTE (Oct-29-2022):
+    // cusparseCreateCsr only supports the same sizes (either 32 bits or 64
+    // bits) for row_map_type and entries_type
+    if (std::is_same<idx_type, int>::value) {
+      if (!std::is_same<size_type, int>::value)
+        sptrsv_handle->allocate_tmp_int_rowmap(row_map.extent(0));
+      rm = !std::is_same<size_type, int>::value
+               ? (void*)sptrsv_handle->get_int_rowmap_ptr_copy(row_map)
+               : (void*)row_map.data();
+    } else {  // idx_type has 64 bits
+      if (!std::is_same<size_type, int64_t>::value)
+        sptrsv_handle->allocate_tmp_int64_rowmap(row_map.extent(0));
+      rm = !std::is_same<size_type, int64_t>::value
+               ? (void*)sptrsv_handle->get_int64_rowmap_ptr_copy(row_map)
+               : (void*)row_map.data();
+    }
+    const scalar_type alpha = scalar_type(1.0);
+
+    cusparseIndexType_t cudaCsrColIndType, cudaCsrRowMapType;
+    cudaDataType cudaValueType;
+
+    if (std::is_same<idx_type, int>::value) {
```

Comment:
This is implemented in [cusparse_index_type_t_from](https://github.com/kokkos/kokkos-kernels/blob/491fbfba3c183c5d7ba2b2eb5fdd1e3a9e2a8bac/sparse/src/KokkosSparse_Utils_cusparse.hpp#L162)

Reply:
@lucbv This was fixed.